python/cuspatial/cuspatial/core/binpreds/binpreds.py

```diff
@@ -168,40 +175,82 @@ def _op(self, lhs, points):
             raise TypeError(
                 "`.contains` can only be called with polygon series."
             )
-
         # call pip on the three subtypes on the right:
-        point_result = contains_properly(
-            points.x,
-            points.y,
-            lhs.polygons.part_offset[:-1],
-            lhs.polygons.ring_offset[:-1],
-            lhs.polygons.x,
-            lhs.polygons.y,
-        )
+        if self.allpairs:
+            point_result = contains_properly_quadtree(
+                points,
+                lhs,
+            )
+        else:
+            point_result = contains_properly_pairwise(
+                points.points.x,
+                points.points.y,
+                lhs.polygons.part_offset,
+                lhs.polygons.ring_offset,
+                lhs.polygons.x,
+                lhs.polygons.y,
+            )
         return point_result
 
     def postprocess(self, point_indices, point_result):
         """Postprocess the output GeoSeries to ensure that they are of the
         correct type for the predicate."""
-        result = cudf.DataFrame({"idx": point_indices, "pip": point_result})
-        df_result = result
-        # Discrete math recombination
-        if (
-            contains_only_linestrings(self.rhs)
-            or contains_only_polygons(self.rhs)
-            or contains_only_multipoints(self.rhs)
-        ):
-            # process for completed linestrings, polygons, and multipoints.
-            # Not necessary for points.
-            df_result = (
-                result.groupby("idx").sum().sort_index()
-                == result.groupby("idx").count().sort_index()
+        if self.allpairs:
+
+            # This complex block of code is to create a dataframe that
+            # contains the polygon index and the point index for each
+            # point in the polygon. Quadtree pip returns the _part_index_
+            # of the polygon, not the polygon index.
+            rings_to_parts = cp.array(self.lhs.polygons.part_offset)
+            part_sizes = rings_to_parts[1:] - rings_to_parts[:-1]
+            parts_map = cudf.Series(
+                cp.arange(len(part_sizes)), name="part_index"
+            ).repeat(part_sizes)
+            parts_df = parts_map.reset_index(drop=True).reset_index()
+            # Mapping of parts to polygons
+            parts_to_geoms = cp.array(self.lhs.polygons.geometry_offset)
+            geometry_sizes = parts_to_geoms[1:] - parts_to_geoms[:-1]
+            geometry_map = cudf.Series(
+                cp.arange(len(geometry_sizes)), name="polygon_index"
+            ).repeat(geometry_sizes)
```

Comment:
https://github.com/rapidsai/cuspatial/pull/910#discussion_r1113698735
```suggestion
            pcol = self.lhs.polygons
            offsets = pcol.ring_offset.take(pcol.part_offset).take(self.geometry_offset)
            sizes = offsets[1:] - offsets[:-1]
            geometry_map = self.Series(cp.arange(len(pcol.xy) // 2).repeat(sizes)
```